contracts/governance/FeeSharingCollector/FeeSharingCollector.sol

```diff
@@ -393,133 +422,261 @@ contract FeeSharingCollector is
+        (totalAmount, endTokenCheckpoint) = _withdraw(_token, _maxCheckpoints, _receiver);
     }
 
-    function _withdrawRBTC(uint32 _maxCheckpoints, address _receiver) internal {
-        uint256 wrbtcAmount;
-        uint256 rbtcAmount;
-        uint256 iWrbtcAmount;
-        uint256 endRBTC;
-        uint256 endWRBTC;
-        uint256 endIWRBTC;
-        uint256 iWRBTCloanAmountPaid;
+    function _withdrawRbtcToken(
+        address _token,
+        uint32 _maxCheckpoints,
+        address _receiver
+    ) internal returns (uint256 totalAmount, uint256 endTokenCheckpoint) {
         address user = msg.sender;
 
         IWrbtcERC20 wrbtcToken = protocol.wrbtcToken();
 
         address loanPoolTokenWRBTC = _getAndValidateLoanPoolWRBTC(address(wrbtcToken));
 
-        if (_receiver == ZERO_ADDRESS) {
-            _receiver = msg.sender;
-        }
-
-        (rbtcAmount, wrbtcAmount, iWrbtcAmount, endRBTC, endWRBTC, endIWRBTC) = _getRBTCBalances(
-            user,
-            _maxCheckpoints
-        );
-
-        if (rbtcAmount > 0) {
-            processedCheckpoints[user][RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT] = endRBTC;
-        }
-
-        // unwrap the wrbtc
-        if (wrbtcAmount > 0) {
-            processedCheckpoints[user][address(wrbtcToken)] = endWRBTC;
-            wrbtcToken.withdraw(wrbtcAmount);
-        }
-
-        // pull out the iWRBTC to rbtc to this feeSharingCollector contract
-        if (iWrbtcAmount > 0) {
-            processedCheckpoints[user][loanPoolTokenWRBTC] = endIWRBTC;
-            iWRBTCloanAmountPaid = ILoanTokenWRBTC(loanPoolTokenWRBTC).burnToBTC(
-                address(this),
-                iWrbtcAmount,
-                false
-            );
+        (totalAmount, endTokenCheckpoint) = _getRBTCBalance(_token, user, _maxCheckpoints);
+
+        if (totalAmount > 0) {
+            processedCheckpoints[user][_token] = endTokenCheckpoint;
+            if (_token == address(wrbtcToken)) {
+                // unwrap the wrbtc
+                wrbtcToken.withdraw(totalAmount);
+            } else if (_token == loanPoolTokenWRBTC) {
+                // pull out the iWRBTC to rbtc to this feeSharingCollector contract
+                /** @dev will use the burned result from IWRBTC to RBTC as return total amount */
+                totalAmount = ILoanTokenWRBTC(loanPoolTokenWRBTC).burnToBTC(
+                    address(this),
+                    totalAmount,
+                    false
+                );
+            }
         }
 
-        uint256 totalAmount = rbtcAmount.add(wrbtcAmount).add(iWRBTCloanAmountPaid);
-        require(totalAmount > 0, "FeeSharingCollector::withdrawFees: no rbtc for a withdrawal");
-
-        // withdraw everything
-        (bool success, ) = _receiver.call.value(totalAmount)("");
-        require(success, "FeeSharingCollector::withdrawRBTC: Withdrawal failed");
-
         emit RBTCWithdrawn(user, _receiver, totalAmount);
```

Comment:
1) pls move to the actual RBTC sending  
2) `_receiver` param is redundant in this function

Reply:
updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/507/commits/efb8c7eef560dc995d949234265110c21647bded)